content/docs/Index-Structure.md

```diff
@@ -0,0 +1,130 @@
++++
+author = ["Patrick Mézard"]
+date = "2015-10-09T00:00:00+01:00"
+title = "Index Structure"
+[menu.docs]
+weight = 30
+parent = 'devguide'
+
++++
+
+Bleve default index, called "upside_down" is stored in a single key/value store
+table. The store must be able to enumerate its entries starting at a given key,
+in lexicographic byte order.
+
+Index key and values are handled as byte arrays and are called rows (see `index/upside_down/row.go` for details). To store different row types in a single table, bleve prefixes their keys with a single byte, for instance the term frequency keys start with a 'f'. The following sections describe the data structures stored in the index with pseudo-Go code for value layout.
+
+### Version Row
+
+Key: `"v"`
+
+Value:
+```
+struct {
+	Version uint8
+}
+```
+
+Only one version row exists in the index and is used to check version compatibility
+
+
+### Field Rows
+
+Key: (`"f"`, `fieldIndex`)
+
+Value:
+```
+struct {
+	Name string
+}
+```
+
+They map field qualified names to internal integer indices. The field names are only used or displayed at API level, bleve uses the indices everywhere internally.
+
+
+### Dictionary Rows
+
+Key: (`"d"`, `fieldIndex`, `term`)
+
+Value:
+```
+struct {
+	Count uint64
+}
+```
+
+Dictionary rows index (field, term) pairs and maintain their number of occurrences. Range queries over terms of a given field, which includes prefix queries, are implemented by seeking over dictionary rows first.
+
+
+### Term Frequency Rows
+
+Key: (`"t"`, `fieldIndex`, `term`, `docId`)
+
+Value:
+```
+type TermFrequencyRow struct {
+	Freq    uint64
+	Norm    float32
+	Vectors []struct{
+		Field          uint16
+		ArrayPositions []uint64
+		Pos            uint64
+		Start          uint64
+		End            uint64
+	}
+}
+```
+
+Term frequency rows store statistics about a term in a document field. Term occurrences can be recorded as well.
+
+`Freq` is the number of occurrences of the term in this field.
+`Norm` is 1/sqrt(number of tokens in this field).
```

Comment:
For further background, the norm is a factor used to help balance scoring of shorter documents which naturally have lower term frequencies.  The value is use at scoring time, but can be computed now at index time.  Future versions may make normalization optional, currently it is not.